src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -1,26 +1,36 @@
 package uk.gov.justice.digital.hmpps.approvedpremisesapi.service
 
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.ClientResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.CommunityApiClient
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserEntity
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRepository
 import java.util.UUID
 
 @Service
 class UserService(
   private val httpAuthService: HttpAuthService,
+  private val communityApiClient: CommunityApiClient,
   private val userRepository: UserRepository
 ) {
   fun getUserForRequest(): UserEntity {
     val deliusPrincipal = httpAuthService.getDeliusPrincipalOrThrow()
     val username = deliusPrincipal.name
 
-    return userRepository.findByDistinguishedName(username)
+    val staffUserDetailsResponse = communityApiClient.getStaffUserDetails(username)
+
+    val staffUserDetails = when (staffUserDetailsResponse) {
+      is ClientResult.Success -> staffUserDetailsResponse.body
+      is ClientResult.Failure -> staffUserDetailsResponse.throwException()
+    }
+
+    return userRepository.findByDeliusUsername(username)
       ?: userRepository.save(
         UserEntity(
           id = UUID.randomUUID(),
-          name = username,
-          distinguishedName = username,
-          isActive = true,
+          name = "${staffUserDetails.staff.forenames} ${staffUserDetails.staff.surname}",
+          deliusUsername = username,
+          deliusStaffIdentifier = staffUserDetails.staffIdentifier,
           applications = mutableListOf()
         )
       )
```

Comment:
I was expecting that `communityApiClient.getStaffUserDetails(username)` would only be called if `userRepository.findByDeliusUsername(username)` came back blank. Probably I'm not reading this properly?

I see the unit tests in `UserServiceTest` and I wonder if they'd be clearer if they described the way the service interacts with the repository rather than asserting what comes back from the repo, which I think is stubbed in any case?

e.g.

- the test `getUserForRequest returns existing User when exists` asserts that the repo is NOT sent a `save()` message but only a `findByDeliusUsername()` message, and 

- the test `getUserForRequest returns new User when one does not already exist` asserts that the repo IS sent a `save()` as well as the `findByDeliusUsername()` message

Reply:
You're quite right, end of day brain haze 🥴 Have amended!